src/server/editorServices.ts

```diff
@@ -873,9 +873,11 @@ namespace ts.server {
             this.delayEnsureProjectForOpenFiles();
         }
 
-        private delayUpdateProjectGraphs(projects: readonly Project[]) {
+        private delayUpdateProjectGraphs(projects: readonly Project[], clearSourceMapperCache?: boolean) {
             if (projects.length) {
                 for (const project of projects) {
+                    // Even if program doesnt change, clear the source mapper cache
+                    if (clearSourceMapperCache) project.clearSourceMapperCache();

```

Comment:
We need to do this for correctness so its question of accuracy.. Note that from perf perspective things are still reused since we watch .map files, .d.ts files etc and use the script info to track it.. So its contents and snapshots etc are updated only on change.. This ensures that if .d.ts file has changed, it remaps the mappings correctly..